ui/src/components/ConstrainedText.js

```diff
@@ -0,0 +1,28 @@
+//@flow
+import React from 'react';
+import type { Node } from 'react';
+import { Tooltip } from '@scality/core-ui';
+import styled from 'styled-components';
+
+const ConstrainedTextContainer = styled.div`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  word-wrap: break-word;
+  white-space: nowrap;
+`;
+
+const BlockTooltip = styled.div`
+  & > .sc-tooltip {
+    display: block;
+  }
+`;
+
+export function ConstrainedText({ text }: { text: string }): Node {
```

Comment:
I was about to move this component to core ui, maybe we should do this before merging this PR

Reply:
will do that!